catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpointTest.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    @Test
+    public void testReportWithFederationAdminException() throws FederationAdminException {
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(),
+                anyList())).thenThrow(new FederationAdminException());
+        Response response = restEndpoint.viewRegInfoHtml("metacardId", Arrays.asList("sourceIds"));
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+    }
+
+    @Test
+    public void testReportRegistryPackageNotFound() throws FederationAdminException {
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(),
+                anyList())).thenReturn(null);
+        Response response = restEndpoint.viewRegInfoHtml("metacardId", Arrays.asList("sourceIds"));
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.NOT_FOUND.getStatusCode()));
+    }
+
+    @Test
+    public void testReportWithRegistryPackage() throws ParserException, FederationAdminException {
+        when(federationAdminService.getRegistryObjectByRegistryId(anyString(),
+                anyList())).thenReturn(registryPackage);
+
+        Response response = restEndpoint.viewRegInfoHtml("metacardId", Arrays.asList("sourceIds"));
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.OK.getStatusCode()));
+
+    }
+
+    @Test
+    public void testPublishMissingFirstArg() throws Exception {
+        Response response = restEndpoint.publish(null, DESTINATION_ID);
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.BAD_REQUEST.getStatusCode()));
+    }
+
+    @Test
+    public void testPublishMissingSecondArg() throws Exception {
+        Response response = restEndpoint.publish(REGISTRY_ID, null);
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.BAD_REQUEST.getStatusCode()));
+    }
+
+    @Test
+    public void testPublishHelperException() throws Exception {
+        doThrow(new FederationAdminException("error")).when(registryPublicationService)
+                .publish(REGISTRY_ID, DESTINATION_ID);
+
+        Response response = restEndpoint.publish(REGISTRY_ID, DESTINATION_ID);
+        verify(registryPublicationService).publish(REGISTRY_ID, DESTINATION_ID);
+
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.BAD_REQUEST.getStatusCode()));
+    }
+
+    @Test
+    public void testPublish() throws Exception {
+        Response response = restEndpoint.publish(REGISTRY_ID, DESTINATION_ID);
+        verify(registryPublicationService).publish(REGISTRY_ID, DESTINATION_ID);
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.OK.getStatusCode()));
+    }
+
+    @Test
+    public void testUnpublishMissingFirstArg() throws Exception {
```

Comment:
Recommend re-naming these two methods to something like `testUnpublishNullRegistryId`. Might be a little more clear.